core/src/main/java/io/parsingdata/metal/data/Slice.java

```diff
@@ -31,24 +34,29 @@
     public final long offset;
     public final BigInteger length;
 
-    public Slice(final Source source, final long offset, final BigInteger length) {
+    private Slice(final Source source, final long offset, final BigInteger length) {
         this.source = checkNotNull(source, "source");
         this.offset = offset;
         this.length = checkNotNull(length, "length");
     }
 
+    public static Optional<Slice> createFromSource(final Source source, final long offset, final BigInteger length) {
+        if (length.compareTo(ZERO) < 0 || !source.isAvailable(offset, length)) { return Optional.empty(); }
+        return Optional.of(new Slice(source, offset, length));
+    }
+
+    public static Slice createFromBytes(final byte[] data) {
+        return new Slice(new ConstantSource(data), 0, BigInteger.valueOf(data.length));
```

Comment:
You *could* call `createFromSource` here...

Reply:
Agreed, but I don't want to have this dirty automatic `.get()` added to it at several places.